src/common/config_parser.cpp

```diff
@@ -679,16 +696,25 @@ void ConfigParser::addOptionsTranslation(cli::CLIWrapper& cli) {
       "Keep the output segmented into SentencePiece subwords");
 #endif
 
-  addSuboptionsInputLength(cli);
-  addSuboptionsTSV(cli);
-  addSuboptionsDevices(cli);
-  addSuboptionsBatching(cli);
+  if(mode_ != cli::mode::selfadaptive) {
+    addSuboptionsInputLength(cli);
+    addSuboptionsTSV(cli);
+    addSuboptionsDevices(cli);
+    addSuboptionsBatching(cli);
+  } else {
+    cli.add<size_t>("--max-length-translate",
+        "Maximum input sentence length for translation",
+        1000);
```

Comment:
Can you comment why it is here? Is it because by combining training and decoding options in marian-adaptive, both options have the same name and one couldn't be used? Consider defining this option inside `addSuboptionsInputLengt` if possible.

Reply:
Explained it in a comment. It's because the use rmight want a different value for training and translation in self-adaptive mode.